File: LibraryManagementSystemBackend/Services/BookService.cs
using LibraryManagementSystemBackend.Data;
using LibraryManagementSystemBackend.Models;
using Microsoft.EntityFrameworkCore;

namespace LibraryManagementSystemBackend.Services
{
    public class BookService : IBookService
    {
        private readonly LibraryDb _context;

        public BookService(LibraryDb context)
        {
            _context = context;
        }

        public async Task<Book> AddBook(Book book)
        {
            _context.Books.Add(book);
            await _context.SaveChangesAsync();
            return book;
        }

        public async Task<IEnumerable<Book>> GetAllBooks()
        {
            return await _context.Books.ToListAsync();
        }

        public async Task<Book?> GetBookById(int id)
        {
            return await _context.Books.FindAsync(id);           
        }

        public async Task<Book?> UpdateBook(int id, Book book)
        {
            try
            {
           var  currentBook = await _context.Books.FindAsync(id);

           if(currentBook == null)
            {
                return null;
            }

            currentBook.Title = book.Title;
            currentBook.Isbn = book.Isbn;
            currentBook.Author = book.Author;
            currentBook.IsAvailable = book.IsAvailable;
            currentBook.PublicationYear = book.PublicationYear;
            currentBook.Description = book.Description;

            await _context.SaveChangesAsync();
            return currentBook;
            }
            catch (DbUpdateException e)
            {
                throw new Exception("Database update error: " + e.Message);
            }
          
        }

        public async Task<Book?> DeleteBook(int id)
        {
            var book = await _context.Books.FindAsync(id);
            if(book == null)
            {
                return null;
            }

            _context.Books.Remove(book);
            await _context.SaveChangesAsync();
            return book;
        }
    }
}

File: LibraryManagementSystemFrontend/src/services/bookService.ts
import axios from "axios";
import type { Book } from "../models/Book";

const API_URL = 'http://localhost:5089/api/Book';

export const bookService = {

    getAllBooks : () => axios.get<Book[]>(API_URL),
    getBookById : (id : number) => axios.get<Book>(`${API_URL}/${id}`),
    createBook : (book : Book) => axios.post<Book>(API_URL, book),
    updateBook : (id : number, book : Book) => axios.put<Book>(`${API_URL}/${id}`, book),
    deleteBook : (id: number)=> axios.delete(`${API_URL}/${id}`)
};